### DISCUSSION TITLE
Using the C++ Parser as a library

### DISCUSSION THREAD

**Discussion by BenHanson** [GregUtas/robust-services-core#175] [Ideas]
Hello,

I would like to use the C++ Parser from this project as a standalone library. Is this a use case that interests you?

If it were possible, I would be interested in contributing code to improve the parser.

**Comment by GregUtas** [comment]
Hi Ben,

The parser wasn't designed with a standalone library in mind, so you'd probably have to clone it and do some decoupling. You're welcome to improve it, but the new things that it supports should also be "compiled"--that is, understood to the point where static analysis can be applied to the code that was parsed.
